src/Draco.Compiler/Api/CodeCompletion/SignatureService.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Draco.Compiler.Api.Semantics;
+using Draco.Compiler.Api.Syntax;
+
+namespace Draco.Compiler.Api.CodeCompletion;
+
+/// <summary>
+/// Provides <see cref="SignatureItem"/>s.
+/// </summary>
+public static class SignatureService
```

Comment:
I know that currently this makes sense to have as a singleton, but let's turn this into an object. We might want to add filters and such later to this, in which case a singleton will rip the idea apart